apps/web/components/Editor/Editor.vue

```diff
@@ -15,41 +15,38 @@
         class="w-full p-2 font-mono text-sm border-none resize-none outline-none h-[500px]"
         :placeholder="$t('editMode.editJsonPlaceholder')"
       ></textarea>
+      <SfButton size="sm" class="absolute top-2 right-2" @click="closeEditor"
+        ><SfIconCancel class="cursor-pointer"></SfIconCancel
+      ></SfButton>
     </div>
     <div v-if="errorMessage" class="text-red-500 mt-2 text-sm">{{ errorMessage }}</div>
-    <UiButton @click="formatJson" class="mt-4 px-4 py-2 text-white rounded-md bg-blue-500 hover:bg-blue-600">
-      {{ $t('editMode.formatJson') }}
-    </UiButton>
-    <UiButton @click="purgeJson" class="mt-4 ml-2 px-4 py-2 text-white rounded-md bg-green-500 hover:bg-green-600">
-      {{ $t('editMode.minifyJson') }}
-    </UiButton>
-    <UiButton @click="clearText" class="mt-4 ml-2 px-4 py-2 text-white rounded-md bg-red-500 hover:bg-red-600">
-      {{ $t('editMode.clearJson') }}
-    </UiButton>
   </div>
 </template>
 
 <script setup lang="ts">
-const { data } = useHomepage();
+import { SfIconCancel } from '@storefront-ui/vue';
 
-const {
-  jsonText,
-  errorMessage,
-  lineCount,
-  textarea,
-  lineNumberContainer,
-  syncScroll,
-  handleInput,
-  formatJson,
-  purgeJson,
-  clearText,
-} = useJsonEditor(JSON.stringify(data, null, 2));
+const props = defineProps<{
+  block: Block | null;
+  index: number;
+}>();
+
+const emit = defineEmits(['update']);
```

Comment:
**suggestion**: Emitting an event shouldn't be necessary if the editor modifies the state (see below).

Reply:
Will be addressed separately.